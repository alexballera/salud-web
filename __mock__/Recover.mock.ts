export const wrapperOptions = {
  formikProps: {
    initialValues: {
      email: '',
      pinCode: '',
      validPin: '0',
      newPassword: '',
      newPasswordConfirm: ''
    }
  }
};
