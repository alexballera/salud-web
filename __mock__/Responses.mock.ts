export const StandardOk = {
  status: 200,
  statusText: 'Ok',
  headers: {},
  config: {}
};

export const LoginEmptyResultResponse = {
  ...StandardOk,
  data: { result: {} }
};
