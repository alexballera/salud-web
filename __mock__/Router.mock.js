export const useRouterMock = () => {
  const back = jest.fn()

  return {
    back
  }
}
