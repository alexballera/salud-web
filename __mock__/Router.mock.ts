export const useRouterMock = (): { back: any } => {
  const back = jest.fn();

  return {
    back
  };
};
