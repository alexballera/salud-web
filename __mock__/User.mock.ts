export const ReturnedUser = {
  userId: 499,
  userId_Ref: 'd85eb911-3e24-4021-8ec6-3bca9c8d31dc',
  fullName: 'Cognito Carlos 70 Cognito70',
  firstName: 'Cognito Carlos 70',
  secondName: 'Cognito70',
  documentType: 1,
  documentNumber: 'V146249870',
  genderId: 1,
  gender: 'Masculino',
  birthDate: '1990-01-01T00:00:00.000Z',
  country: 'Costa Rica',
  city: 'San Jose',
  region: 'San Jose',
  address: 'San Jose',
  zipCode: '90011',
  medical_License: null,
  isMedic: {
    type: 'Buffer',
    data: [0]
  },
  isPatient: {
    type: 'Buffer',
    data: [1]
  },
  status: 'Active',
  statusCode: 1,
  createdAt: '2021-05-27T21:02:09.000Z',
  updatedAt: '2021-05-27T21:02:25.000Z',
  emailId: 459,
  email: 'ftgdloebo@logicstreak.com',
  codSAC: 241,
  phoneNumber: '123456791',
  firstMedicalRecordComplete: 0,
  token: {
    access_token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6Ijk5NGJkNzQ2LTUxN2EtNDAxNi1hZjRiLWVhNjk2YzAyYTUxYiJ9.eyJpc3MiOiJodHRwczpcL1wvd3d3LnNhbHVkYXVuY2xpYy5jb21cLyIsImp0aSI6Ijk5NGJkNzQ2LTUxN2EtNDAxNi1hZjRiLWVhNjk2YzAyYTUxYiIsImlhdCI6MTYyMjY0MTkxMCwibmJmIjoxNjIyNjQxOTEwLCJleHAiOjE2MjI2NTYzMTAsImRhdGEiOiJ7XCJzYWNJZFwiOlwiMjQxXCIsXCJpbnN0aXR1dGlvbklkXCI6XCIxM1wiLFwidXNlcklkXCI6XCJWMTQ2MjQ5ODcwXCIsXCJ1c2VySWRUeXBlXCI6XCIxXCIsXCJkYXRlT2ZCaXJ0aFwiOlwiMTk5MC0wMS0wMVwiLFwiZ2VuZGVyXCI6XCJNYWxlXCIsXCJsYXN0U3VybmFtZVwiOm51bGwsXCJuYW1lXCI6XCJDb2duaXRvIENhcmxvcyA3MFwiLFwic3VybmFtZVwiOlwiQ29nbml0bzcwXCIsXCJlbWFpbFwiOlwiZnRnZGxvZWJvQGxvZ2ljc3RyZWFrLmNvbVwiLFwicGhvbmVcIjpcIjEyMzQ1Njc5MVwiLFwicHdkVG1wXCI6XCIwXCJ9In0.Yqnp_p1M7GFoHawp4Rprvre8aXFpHD_amAzgRe-EG8cjI98w52J-Ql0ptivb6lioAeBwUhLROoE0IT8V6NFvLLIeonxSJAR0C8vc69PbVEUlXQQYrSmcQ2cp5wgeUsiYL4h-1G4jfkAmxkWRG1nKzW70EgpODd6uXgxvSrzcRv2FwSaaJNwF5ciCYZzx8Okql8nnm_d1jY39GOyPqtsnobxpiYdNi9XNMGeERs7QY-tH48k9_d0vqLkkOiDvOI-EYhx9hrEy_fsEXbZQ8Y5_GUoFvMk2Dz7IzoPDdFSvZvAp2vLU_YZMXbUxxiAlUn_CF-PCcZy3plByaKiaZfrUPA',
    token_type: 'Bearer',
    expires_in: 14400
  },
  cognitoToken: {
    accessToken:
      'eyJraWQiOiI1aWtpVlA1WDVZMnREZW43cEdmNkpGbXY3cllvUzJ6WUFGMXY3aEpzNGtJPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiJkODVlYjkxMS0zZTI0LTQwMjEtOGVjNi0zYmNhOWM4ZDMxZGMiLCJldmVudF9pZCI6ImE0YTE5YjZlLWY3YTktNDAzNS04MTNjLWY5N2E3MGJkYjVjYiIsInRva2VuX3VzZSI6ImFjY2VzcyIsInNjb3BlIjoiYXdzLmNvZ25pdG8uc2lnbmluLnVzZXIuYWRtaW4iLCJhdXRoX3RpbWUiOjE2MjI2NDE5MTAsImlzcyI6Imh0dHBzOlwvXC9jb2duaXRvLWlkcC51cy1lYXN0LTEuYW1hem9uYXdzLmNvbVwvdXMtZWFzdC0xX2ZFbWFpaHJoRiIsImV4cCI6MTYyMjY0NTUxMCwiaWF0IjoxNjIyNjQxOTEwLCJqdGkiOiJmYmIyNzhmMy02OGYwLTQ2M2ItYTJkZC02MGFiZTEyNDdkYjgiLCJjbGllbnRfaWQiOiIyOXM5YzJhaGpsNTVkYmhvMGM4MmdjMzNhNSIsInVzZXJuYW1lIjoiZDg1ZWI5MTEtM2UyNC00MDIxLThlYzYtM2JjYTljOGQzMWRjIn0.PZU1AsnJFzP-sMGhyRiRUBpNDt3A6pPu6XhQ7bIx7qb21NWm019a6qqh3koc0uQ8dBEm8_H8XYdPcVC1eQUnM1SVAl6hVa-U1uqv-MyxmyZzaxvuVrQNEmYoBd0K13RNQXNAdGFghKdZjjNzIrKAGb94FtyLjNtU0Hh1cMZGKggAedE7Yfi3GD72cU6YUUjwtgEr2XXJvS1u8drI-qkDMXbyMa21pVosxgwnvr746X2dlp4HEmxEBan4--aTwUCNIhnWN7isCtPcnDLsAUv5IH4AL4Jkjq2dx1dY0k2DC6qahnRI2pOi3jGHsilwhTHZFQ0OuQIB-RyNqSrC331yxA',
    expiresIn: 3600,
    tokenType: 'Bearer',
    refreshToken:
      'eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAifQ.ULC8WJiXlnZegxd3a8p6kdMaCzZ2e9Ss3hM4UlavWABxPVBV5jy2nTGXE5XV-zCziAwx_fkJdGX2pY1TZ61Sk1bBqyKBK0G3FAWOdAT9EgNNsTQEGvIs6VqgRT8m83elMBnSwiKlOuEZUZTV1oq4eqFrLz-zkD3u2yeCwt3DLEene1bKAQlZ8kSGHB-CPavc4a2_Jur6Pxr4AOxyAzt9YLMGjrBzVj3HP-NSXD2OhEFSTurCAgCaqqeuks_h3j2tQJYyrGlIYTn1z3v9iBZ3RUhMG73aWiwyzGlNlY_dCENLpnPb8ZD3nHB81gDt-r3zSpnnN0_YZNbFVUCBYNeeaQ.u5wl7ZV5MaIJySsE.A-VvLz_BH4GOyCFfaHrgfEZtJi8WHR_eYTDY3ZVZuDyJoB34IbvWaRinabmr9zrk8nqoBghdzWD5OZH9ypfdyRFFwwKgPZUXOTGPf6aWEfxKyEc02OxIgYvk1kYHNo6ePdWFmUjRIcygF7oxTA5e12yGjpLl3UG7RXY1S2PEhC1Z-nRTmg5ogYXczcgzyxowb3frrAYO0zZ-TER371CzdrAx2UvSSlwvaMPYY0wc0jDAbfbKLu2zyogJCJ-T2sJrEg6MTYzrnAsikV6JsRrFMdURaeBYJ_MMo8xhRDIYVKSsF8oWL-IrTwQKLbU0IA7RSuTApNEJybKZxGblm_3yi3CudJ4-dZi-fYyP3QGeK6Oti-TlaGdoojqvRI8U3Rf0xEppdNmodiYoWgAJam5Z2JbiGz4Jm6xAd1nzz7dnXiZlT18zqr7tGZIuIo0kwDrjaKhUd0eqkMyQx6fR8QLednjZPgB9-qWiBv5Zv1iKZDF000U-Rnftl1qpdo8NY48XB6FMxFFQ_XYC47WWVD0pvxoBV8xKCRs0D0OmHhW6553pC9c4NQl8PBDIP4deQQwWEsy9Jqq3JJ6s4M9XUWn2wX7uqVjF0Bht2Iy7ivYUCHhCKuBXky_sDGozG5ZnekyFJWgsZoHQJYGX4D5jUJ4AaumFas8HosvzRcxVhUbSmVJaW14U9RAMuXjfdsbly7mUs36xtlLLOiaN-08V700EQ3Of44ndJvDZbMDl0u_z80GYt5AEe1Cpw1XWB8-w_4JtC7VOA7N2YHILudGqGpjGc-mDSIHigUrg2ui9eHA-BBe_Q_F98flIQIJ8b9xdqWzZ0iXUUTrc5vFxDXWDp3uGZFD8GEqZV24HDqo6TnY17EmjwU4U6RFL8bH-_tnJEZhvo-IpbYdOjBhbN915nJCMEzUE6v6jAumHSk9I1myQQkHbRy1S5Dry6KKZ_S5B_CgsP_3Whn1vZiX2d3syaFt80TdqI0gVOSLIexk8v-XUJbYHE2hqYLPKo0VDidziQkXGUEuKVu-J33sCe2VWmhUm-lzYbhgPIlOZ9WFzIgFORz-gwjgjVwn21V8ME9yTCTkzoP2e9LudBY6sHPUdm7nnKkbyYTyL7lt7Xp5GbKx6nxWumu25lVje580qc0IKpq2f4isSZQ2oVdpYwcG0M_4hgqMPcFv0O1gp86eW3hSH4n0enBgKQV5HJbqxMwLABoEbcQerSmz8sgCnbKlHS_d6rEVadPo_YonajQ5gXGGLM2PJ_EvfnwMSXFsStaPH71fchJEt7S8bA8PRSDjSBt8vAvQksbjoP1UYYDLHTULQReJDT_dOeOZ2hbG7qQ.vBFDDSbxSBhjsVKrjG_srg',
    idToken:
      'eyJraWQiOiJEb1BuRm1wVzEyV2g3Qk1qbXJ0S09WdmxpZmtSVzBvRk5tbVdxV0ZabUJFPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiJkODVlYjkxMS0zZTI0LTQwMjEtOGVjNi0zYmNhOWM4ZDMxZGMiLCJhdWQiOiIyOXM5YzJhaGpsNTVkYmhvMGM4MmdjMzNhNSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJldmVudF9pZCI6ImE0YTE5YjZlLWY3YTktNDAzNS04MTNjLWY5N2E3MGJkYjVjYiIsInRva2VuX3VzZSI6ImlkIiwiYXV0aF90aW1lIjoxNjIyNjQxOTEwLCJpc3MiOiJodHRwczpcL1wvY29nbml0by1pZHAudXMtZWFzdC0xLmFtYXpvbmF3cy5jb21cL3VzLWVhc3QtMV9mRW1haWhyaEYiLCJjb2duaXRvOnVzZXJuYW1lIjoiZDg1ZWI5MTEtM2UyNC00MDIxLThlYzYtM2JjYTljOGQzMWRjIiwiZXhwIjoxNjIyNjQ1NTEwLCJpYXQiOjE2MjI2NDE5MTAsImVtYWlsIjoiZnRnZGxvZWJvQGxvZ2ljc3RyZWFrLmNvbSJ9.S7sSWqd8eZJDafCqYQG0nI9gUb9JgkLz7o5wTu1aY98I1o4oGGn7MP7SqzxA13SG0r_QfHOm_DjiaYYp5vtTA1uuNVp5t-ARnphieoRL71Oep_IsSvqX2e1F614kk0outJgwqZVFdRGTxWraSPEl7jml61GnDOvsG4cus90BqwempWK6e5V-JAdhj97Z-iM4fX3Jwaa4EBrf7_X7tSTZR2DIOBgfHItBdtnofdxuOHCHPtg6-S3CCdwSuNUQRzy3p43ooz6UOhZAhjtJr3I3xdoCUsOt-4v2i-FRNz3gfhv-NzdzRW2Se-5NoSOa5C2Jw1FMERxu2n5YtRBbmnxLQA'
  }
};
