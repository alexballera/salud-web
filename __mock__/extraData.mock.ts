export const mockProvincesResult = {
  primerNivel: [{ codigo: 'provincia', nombre: 'provincia' }]
};

export const mockCantonesResult = {
  segundoNivel: [{ codigo: 'canton', nombre: 'canton' }]
};

export const mockDistrictResult = {
  catalogo: [{ codigo: 'distrito', nombre: 'distrito' }]
};

export const wrapperOptions = {
  formikProps: {
    initialValues: {
      gender: '',
      canton: '',
      province: '',
      district: '',
      mobilePhone1: ''
    }
  },
  componentProps: {}
};
