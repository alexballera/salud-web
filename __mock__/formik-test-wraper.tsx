import { render } from '@testing-library/react';
import { Formik, Form } from 'formik';
/// MATERIAL-UI
import Button from '@material-ui/core/Button';

const AllTheSteps = ({ children, ...props }): JSX.Element => {
  return (
    <Formik {...props.formikProps}>
      {formik => (
        <Form>
          {children({ ...formik, ...props.componentProps })}
          <Button type="submit">Enviar</Button>
        </Form>
      )}
    </Formik>
  );
};

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui, options) => {
  return render(ui, {
    // eslint-disable-next-line react/display-name
    wrapper: props => <AllTheSteps {...props} {...options.wrapperOptions} />,
    ...options
  });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
