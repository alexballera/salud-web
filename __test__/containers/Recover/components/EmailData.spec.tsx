import { axe, toHaveNoViolations } from 'jest-axe';
import EmailDataForm from '../../../../src/containers/Recover/components/EmailData';
import { act, fireEvent, render, screen } from '../../../../__mock__/formik-test-wraper';
import { wrapperOptions as defaultWrapperOptions } from '../../../../__mock__/Recover.mock';

expect.extend(toHaveNoViolations);

const wrapperOptions = {
  ...defaultWrapperOptions,
  validationSchema: EmailDataForm.validations
};

describe('<EmailDataForm />', () => {
  it('should render without throwing an error', () => {
    let tree;
    act(() => {
      tree = render(EmailDataForm, { wrapperOptions });
    });

    expect(tree).toMatchSnapshot();
  });

  it('should show error on invalid email', async () => {
    let tree;
    let inputContainer: HTMLElement;
    let submitButton: HTMLElement;
    const submitMock = jest.fn();
    await act(async () => {
      tree = render(EmailDataForm, {
        wrapperOptions: {
          ...wrapperOptions,
          formikProps: { ...wrapperOptions.formikProps, onSubmit: submitMock }
        }
      });

      inputContainer = await screen.findByTestId('email-input');
      submitButton = await screen.findByText('Enviar');

      fireEvent.change(inputContainer.getElementsByTagName('input')[0], {
        target: { value: 'nonEmailValue' }
      });
      fireEvent.click(submitButton);
    });
  });

  it('should submit with valid email', async () => {
    let inputContainer: HTMLElement;
    let submitButton: HTMLElement;
    const submitMock = jest.fn();
    await act(async () => {
      render(EmailDataForm, {
        wrapperOptions: {
          ...wrapperOptions,
          formikProps: { ...wrapperOptions.formikProps, onSubmit: submitMock }
        }
      });

      inputContainer = await screen.findByTestId('email-input');
      submitButton = await screen.findByText('Enviar');

      fireEvent.change(inputContainer.getElementsByTagName('input')[0], {
        target: { value: 'some@email.com' }
      });
      fireEvent.click(submitButton);
    });
    expect(inputContainer.textContent).toEqual('Correo electrónico​');
    expect(inputContainer.textContent).not.toMatch(/Formato de correo incorrecto/);
    expect(submitMock).toBeCalled();
  });

  it('should be accesible', async () => {
    let results;

    await act(async () => {
      render(props => <EmailDataForm {...props} />, {
        wrapperOptions: wrapperOptions
      });

      results = await axe(document.body);
    });

    expect(results).toHaveNoViolations();
  });
});
