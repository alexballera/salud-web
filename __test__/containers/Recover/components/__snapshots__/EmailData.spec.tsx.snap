// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailDataForm /> should render without throwing an error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
         
        <form
          action="#"
        >
          <div
            class="MuiBox-root MuiBox-root-1"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
              data-testid="email-input"
            >
              <label
                class="MuiFormLabel-root"
                for="email"
                margin="normal"
                style="margin-bottom: 10px;"
              >
                Correo electrónico
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-3"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Enviar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
       
      <form
        action="#"
      >
        <div
          class="MuiBox-root MuiBox-root-1"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
            data-testid="email-input"
          >
            <label
              class="MuiFormLabel-root"
              for="email"
              margin="normal"
              style="margin-bottom: 10px;"
            >
              Correo electrónico
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-3"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Enviar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
